backend_test.go

```diff
@@ -28,3 +34,172 @@ func getTestBackend(tb testing.TB) (*tfBackend, logical.Storage) {
 
 	return b.(*tfBackend), config.StorageView
 }
+
+var runAcceptanceTests = os.Getenv(envVarRunAccTests) == "1"
+
+type testEnv struct {
+	Token        string
+	Organization string
+	TeamID       string
+	UserID       string
+
+	Backend logical.Backend
+	Context context.Context
+	Storage logical.Storage
+
+	// SecretToken tracks the API token, for checking rotations
+	SecretToken string
+
+	// TokenIDs tracks the IDs of generated tokens, to make sure we clean up
+	TokenIDs []string
+}
+
+func (e *testEnv) AddConfig(t *testing.T) {
+	req := &logical.Request{
+		Operation: logical.CreateOperation,
+		Path:      "config",
+		Storage:   e.Storage,
+		Data: map[string]interface{}{
+			"token": e.Token,
+		},
+	}
+	resp, err := e.Backend.HandleRequest(e.Context, req)
+	assert.False(t, (err != nil || (resp != nil && resp.IsError())), fmt.Sprintf("bad: resp: %#v\nerr:%v", resp, err))
```

Comment:
Is there a way we could clean this conditional up? We're asserting `false` and then doing a bunch of `!=` which makes it... a bit opaque.

Reply:
Looking at it a bit more in-depth: I THINK this is having it return `"bad: resp: %#v\nerr:%v"` `if err == nil && !(resp != nil && resp.IsError())` or in a further simplified form `if err == nil && (resp == nil || !resp.IsError())`

This is based off of the assessment that the assert.False conditional is equivalent to:
`assert.True(t, !(err != nil || (resp != nil && resp.IsError())), ...`

I'm not positive of that assessment, but it may be worth double checking.

Reply:
So I looked harder at this code and I think it's a bit superfluous. I updated instances where I found it to change the code to be simple `assert.Nil` or `assert.NotNil` checks on the `err` and `resp` objects. I pushed daf31d2 with the changes, let me know what you think!